x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/aggregate/Percentile.java

```diff
@@ -41,13 +42,17 @@ public Percentile replaceChildren(List<Expression> newChildren) {
 
     @Override
     protected TypeResolution resolveType() {
-        TypeResolution resolution = super.resolveType();
+        if (!percent.foldable()) {
+            throw new SqlIllegalArgumentException("2nd argument: [{}] of PERCENTILE cannot be based on " +
```

Comment:
I would rephrase this to something shorter such as "must be a constant, received [{}]".